query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -52,6 +58,16 @@ public String convert(
     }
   }
 
+  private String functionToStringForAvgRate(
```

Comment:
@kotharironak @rish691 I need executionContext here. What is the best way to have it here as by passing it as an argument many function signatures have to be changed.